etcdserver/api/membership/cluster.go

```diff
@@ -778,6 +796,36 @@ func clusterVersionFromBackend(lg *zap.Logger, be backend.Backend) *semver.Versi
 	return semver.Must(semver.NewVersion(string(vals[0])))
 }
 
+func downgradeInfoFromBackend(lg *zap.Logger, be backend.Backend) *DowngradeInfo {
+	dkey := backendDowngradeKey()
+	if be != nil {
+		tx := be.ReadTx()
+		tx.Lock()
+		defer tx.Unlock()
+		keys, vals := tx.UnsafeRange(clusterBucketName, dkey, nil, 0)
+		if len(keys) == 0 {
+			return nil
+		}
+
+		if len(keys) != 1 {
+			if lg != nil {
```

Comment:
Also, do we really need `nil` checks on logger?

Reply:
I don't quite get it. Since we haven't deprecated plog and used zap log as default, lg is set to nil as default if flag `--logger=zap` is not enabled. I think we can remove all the nil check of log after we enable zap log as default?

Reply:
I meant, if we set logger to https://godoc.org/go.uber.org/zap#NewNop once in the caller, all the downstream calls need not handle this `nil` checks?